miniapps/navier/navier_3dfoc.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "navier_solver.hpp"
+#include <fstream>
+
+using namespace mfem;
+using namespace navier;
+
+struct s_NavierContext
+{
+   int order = 4;
+   double kin_vis = 0.001;
+   double t_final = 8.0;
+   double dt = 1e-3;
+} ctx;
+
+void vel(const Vector &x, double t, Vector &u)
+{
+   double xi = x(0);
+   double yi = x(1);
+   double zi = x(2);
+
+   double U = 2.25;
+
+   if (xi <= 1e-8)
+   {
+      u(0) = 16.0 * U * yi * zi * sin(M_PI * t / 8.0) * (0.41 - yi)
+             * (0.41 - zi) / pow(0.41, 4.0);
+   }
+   else
+   {
+      u(0) = 0.0;
+   }
+   u(1) = 0.0;
+   u(2) = 0.0;
+}
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi(argc, argv);
+
+   int serial_refinements = 0;
+
+   Mesh *mesh = new Mesh("../miniapps/navier/3dfoc.e");
```

Comment:
Missing mesh file.

Reply:
Can you use standard mesh or gmsh mesh format? It will simplify the utilization.

Reply:
I cannot see the mesh file 3dfoc.e